components/clp-package-utils/clp_package_utils/scripts/start_clp.py

```diff
@@ -1138,195 +33,37 @@ def main(argv):
-        check_dependencies()
-    except:
-        logger.exception("Dependency checking failed.")
-        return -1
 
     # Validate and load config file
     try:
         config_file_path = pathlib.Path(parsed_args.config)
         clp_config = load_config_file(config_file_path, default_config_file_path, clp_home)
 
-        runnable_components = clp_config.get_runnable_components()
-        components_to_start = get_components_for_target(target)
-        components_to_start = components_to_start.intersection(runnable_components)
-
-        # Exit early if no components to start
-        if len(components_to_start) == 0:
-            logger.error(f"{target} not available with current configuration")
-            return -1
-
-        # Validate and load necessary credentials
-        if target in (
-            ALL_TARGET_NAME,
-            CONTROLLER_TARGET_NAME,
-            DB_COMPONENT_NAME,
-            GARBAGE_COLLECTOR_COMPONENT_NAME,
-            COMPRESSION_SCHEDULER_COMPONENT_NAME,
-            QUERY_SCHEDULER_COMPONENT_NAME,
-            WEBUI_COMPONENT_NAME,
-        ):
-            validate_and_load_db_credentials_file(clp_config, clp_home, True)
-        if target in (
-            ALL_TARGET_NAME,
-            CONTROLLER_TARGET_NAME,
-            QUEUE_COMPONENT_NAME,
-            COMPRESSION_SCHEDULER_COMPONENT_NAME,
-            QUERY_SCHEDULER_COMPONENT_NAME,
-            COMPRESSION_WORKER_COMPONENT_NAME,
-            QUERY_WORKER_COMPONENT_NAME,
-        ):
-            validate_and_load_queue_credentials_file(clp_config, clp_home, True)
-        if target in (
-            ALL_TARGET_NAME,
-            CONTROLLER_TARGET_NAME,
-            REDIS_COMPONENT_NAME,
-            COMPRESSION_SCHEDULER_COMPONENT_NAME,
-            QUERY_SCHEDULER_COMPONENT_NAME,
-            COMPRESSION_WORKER_COMPONENT_NAME,
-            QUERY_WORKER_COMPONENT_NAME,
-        ):
-            validate_and_load_redis_credentials_file(clp_config, clp_home, True)
-        if target in (
-            ALL_TARGET_NAME,
-            COMPRESSION_WORKER_COMPONENT_NAME,
-        ):
-            validate_logs_input_config(clp_config)
-        if target in (
-            ALL_TARGET_NAME,
-            COMPRESSION_WORKER_COMPONENT_NAME,
-            QUERY_WORKER_COMPONENT_NAME,
-            GARBAGE_COLLECTOR_COMPONENT_NAME,
-        ):
-            validate_output_storage_config(clp_config)
-        if target in (
-            ALL_TARGET_NAME,
-            CONTROLLER_TARGET_NAME,
-            GARBAGE_COLLECTOR_COMPONENT_NAME,
-        ):
-            validate_retention_config(clp_config)
+        validate_and_load_db_credentials_file(clp_config, clp_home, True)
+        validate_and_load_queue_credentials_file(clp_config, clp_home, True)
+        validate_and_load_redis_credentials_file(clp_config, clp_home, True)
+        validate_logs_input_config(clp_config)
+        validate_output_storage_config(clp_config)
+        validate_retention_config(clp_config)
```

Comment:
we could have moved all those into the constructor of `BaseController` if we can get rid of the `default_config_file_path` argument and related handling inside `load_config_file()`.